File: hw-2B/A-equal-max.py
def equal_to_max(arr):

    cnt = {}
    for item in arr:
        if item not in cnt:
            cnt[item] = 0
        cnt[item] += 1
    max_el = max(list(cnt.keys()))
    return cnt[max_el]


if __name__ == '__main__':

    "Calculate the number of values in a sequence, which are equal to the maximum value"

    nums = []
    while True:
        x = int(input())
        if x != 0:
            nums.append(x)
        else:
            break
    res = equal_to_max(nums)
    print(res)

File: hw-2B/C++/A-equal-max.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

int equalToMax(vector <int> arr){

    unordered_map <int, int> cnt;
    for (auto item: arr)
        cnt[item] ++;
    
    vector <int>::iterator idx;
    idx = max_element(arr.begin(), arr.end()); 
    // return iterator => use *idx 
    return cnt[*idx];
}

int main(){

    // Calculate the number of values in a sequence, which are equal to the maximum value

    vector <int> nums;
    int x;
    while (true){
        cin >> x;
        if (x != 0)
            nums.push_back(x);
        else
            break;
    }

    int res = equalToMax(nums);
    cout << res << endl;

}